### ISSUE TITLE
I Can't upgrade the istio cluster on k8s And Delete the istio cluster

### ISSUE THREAD

**Issue by tanjunchen** [banzaicloud/istio-operator#434]
**Describe the bug**

Fisrtly , I install the istio cluster by 
```
root@k8s-master:/home/k8s-master/istio-operator# git branch
* release-1.4
  release-1.5

make delpoy

kubectl apply -n istio-system -f config/samples/istio_v1beta1_istio.yaml

root@k8s-master:/home/k8s-master/istio-operator# kubectl get pods -n istio-system
NAME                                     READY   STATUS    RESTARTS   AGE
istio-citadel-584fd7474d-2n648           1/1     Running   0          2m27s
istio-egressgateway-58b98fb485-m6xr9     1/1     Running   0          2m1s
istio-galley-65ffbfdd9b-dzf89            1/1     Running   0          2m27s
istio-ingressgateway-77784d69b4-j98xx    1/1     Running   0          2m2s
istio-operator-controller-manager-0      2/2     Running   0          27m
istio-pilot-554b8db67f-bn7hv             2/2     Running   0          2m26s
istio-policy-7fbc8b7785-hrx8x            2/2     Running   1          2m24s
istio-sidecar-injector-9c5d4b794-php2t   1/1     Running   0          2m1s
istio-telemetry-57ccc984b-wsn2m          2/2     Running   1          2m15s


root@k8s-master:/home/k8s-master/istio-operator# git branch 
  release-1.4
* release-1.5
root@k8s-master:/home/k8s-master/istio-operator# make deploy
bin/kustomize build config | kubectl apply -f -
namespace/istio-system unchanged
customresourcedefinition.apiextensions.k8s.io/istios.istio.banzaicloud.io configured
customresourcedefinition.apiextensions.k8s.io/meshgateways.istio.banzaicloud.io configured
customresourcedefinition.apiextensions.k8s.io/remoteistios.istio.banzaicloud.io configured
./scripts/image_patch.sh "config/overlays/specific-manager-version/manager_image_patch.yaml" banzaicloud/istio-operator:0.5.8
bin/kustomize build config/overlays/specific-manager-version | kubectl apply -f -
clusterrole.rbac.authorization.k8s.io/istio-operator-manager-role configured
clusterrole.rbac.authorization.k8s.io/istio-operator-proxy-role unchanged
clusterrolebinding.rbac.authorization.k8s.io/istio-operator-manager-rolebinding configured
clusterrolebinding.rbac.authorization.k8s.io/istio-operator-proxy-rolebinding unchanged
service/istio-operator-controller-manager-metrics-service unchanged
service/istio-operator-controller-manager-service unchanged
statefulset.apps/istio-operator-controller-manager configured

root@k8s-master:/home/k8s-master/istio-operator# kubectl get pods -n istio-system
NAME                                     READY   STATUS        RESTARTS   AGE
istio-citadel-584fd7474d-2n648           1/1     Running       0          8m18s
istio-egressgateway-58b98fb485-m6xr9     0/1     Terminating   0          7m52s
istio-egressgateway-5bb84b6b48-f5qgz     1/1     Running       0          25s
istio-galley-65ffbfdd9b-dzf89            1/1     Running       0          8m18s
istio-ingressgateway-6c9d58c467-96tmr    1/1     Running       0          24s
istio-operator-controller-manager-0      2/2     Running       0          45s
istio-pilot-554b8db67f-bn7hv             2/2     Running       0          8m17s
istio-policy-7fbc8b7785-hrx8x            2/2     Running       1          8m15s
istio-sidecar-injector-9c5d4b794-php2t   1/1     Running       0          7m52s
istio-telemetry-57ccc984b-wsn2m          2/2     Running       1          8m6s


root@k8s-master:/home/k8s-master/istio-operator# kubectl get pods -n istio-system
NAME                                     READY   STATUS    RESTARTS   AGE
istio-citadel-584fd7474d-2n648           1/1     Running   0          9m15s
istio-egressgateway-5bb84b6b48-f5qgz     1/1     Running   0          82s
istio-galley-65ffbfdd9b-dzf89            1/1     Running   0          9m15s
istio-ingressgateway-6c9d58c467-96tmr    1/1     Running   0          81s
istio-operator-controller-manager-0      2/2     Running   0          102s
istio-pilot-554b8db67f-bn7hv             2/2     Running   0          9m14s
istio-policy-7fbc8b7785-hrx8x            2/2     Running   1          9m12s
istio-sidecar-injector-9c5d4b794-php2t   1/1     Running   0          8m49s
istio-telemetry-57ccc984b-wsn2m          2/2     Running   1          9m3s

root@k8s-master:/home/k8s-master/istio-operator# kubectl replace -n istio-system -f config/samples/istio_v1beta1_istio.yaml
istio.istio.banzaicloud.io/istio-sample replaced


root@k8s-master:/home/k8s-master/istio-operator# kubectl get pods -n istio-system -w
NAME                                     READY   STATUS        RESTARTS   AGE
istio-citadel-584fd7474d-2n648           1/1     Terminating   0          9m59s
istio-egressgateway-5bb84b6b48-f5qgz     1/1     Running       0          2m6s
istio-galley-65ffbfdd9b-dzf89            1/1     Terminating   0          9m59s
istio-ingressgateway-6c9d58c467-96tmr    1/1     Running       0          2m5s
istio-operator-controller-manager-0      2/2     Running       0          2m26s
istio-pilot-554b8db67f-bn7hv             2/2     Running       0          9m58s
istio-policy-7fbc8b7785-hrx8x            2/2     Terminating   1          9m56s
istio-sidecar-injector-9c5d4b794-php2t   1/1     Terminating   0          9m33s
istio-telemetry-57ccc984b-wsn2m          2/2     Terminating   1          9m47s
istio-sidecar-injector-9c5d4b794-php2t   0/1     Terminating   0          9m35s
istio-sidecar-injector-9c5d4b794-php2t   0/1     Terminating   0          9m36s
istio-sidecar-injector-9c5d4b794-php2t   0/1     Terminating   0          9m36s
istio-policy-7fbc8b7785-hrx8x            0/2     Terminating   1          10m
istiod-79c677c5c4-wzzb7                  0/1     Pending       0          1s
istiod-79c677c5c4-wzzb7                  0/1     Pending       0          1s
istio-pilot-554b8db67f-bn7hv             2/2     Terminating   0          10m
istio-telemetry-57ccc984b-wsn2m          0/2     Terminating   1          9m55s
istio-telemetry-57ccc984b-wsn2m          0/2     Terminating   1          9m56s
istio-telemetry-57ccc984b-wsn2m          0/2     Terminating   1          9m56s
istiod-79c677c5c4-wzzb7                  0/1     Pending       0          3s
istiod-79c677c5c4-wzzb7                  0/1     ContainerCreating   0          3s
istio-galley-65ffbfdd9b-dzf89            0/1     Terminating         0          10m
istio-galley-65ffbfdd9b-dzf89            0/1     Terminating         0          10m
istio-galley-65ffbfdd9b-dzf89            0/1     Terminating         0          10m
istio-egressgateway-5bb84b6b48-f5qgz     1/1     Terminating         0          2m18s
istiod-79c677c5c4-wzzb7                  0/1     ContainerCreating   0          6s
istio-policy-7fbc8b7785-hrx8x            0/2     Terminating         1          10m
istio-policy-7fbc8b7785-hrx8x            0/2     Terminating         1          10m
istio-pilot-554b8db67f-bn7hv             0/2     Terminating         0          10m
istio-egressgateway-5bb84b6b48-f5qgz     0/1     Terminating         0          2m25s
istio-egressgateway-5bb84b6b48-f5qgz     0/1     Terminating         0          2m26s
istio-egressgateway-5bb84b6b48-f5qgz     0/1     Terminating         0          2m26s
istio-pilot-554b8db67f-bn7hv             0/2     Terminating         0          10m
istio-pilot-554b8db67f-bn7hv             0/2     Terminating         0          10m
istio-citadel-584fd7474d-2n648           0/1     Terminating         0          10m
istio-citadel-584fd7474d-2n648           0/1     Terminating         0          10m
istio-citadel-584fd7474d-2n648           0/1     Terminating         0          10m
istiod-79c677c5c4-wzzb7                  0/1     Running             0          52s
istiod-79c677c5c4-wzzb7                  1/1     Running             0          56s
istio-ingressgateway-6f5b465849-nk8r8    0/1     Pending             0          0s
istio-ingressgateway-6f5b465849-nk8r8    0/1     Pending             0          0s
istio-ingressgateway-6f5b465849-nk8r8    0/1     ContainerCreating   0          1s
istio-ingressgateway-6f5b465849-nk8r8    0/1     ContainerCreating   0          3s
istio-ingressgateway-6f5b465849-nk8r8    0/1     Running             0          5s
istio-ingressgateway-6f5b465849-nk8r8    1/1     Running             0          7s
istio-ingressgateway-6c9d58c467-96tmr    1/1     Terminating         0          3m27s
^Croot@k8s-master:/home/k8s-master/istio-operator# kubectl get pods -n istio-system 
NAME                                    READY   STATUS        RESTARTS   AGE
istio-ingressgateway-6c9d58c467-96tmr   0/1     Terminating   0          3m35s
istio-ingressgateway-6f5b465849-nk8r8   1/1     Running       0          15s
istio-operator-controller-manager-0     2/2     Running       0          3m56s
istiod-79c677c5c4-wzzb7                 1/1     Running       0          84s

it is failed to upgrade the istio cluster. 

```

**Steps to reproduce the issue:**
Please describe the steps to reproduce the issue.

**Expected behavior**

**Screenshots**

**Additional context**
Add any other context about the problem like release numberm version, branch, etc.

**Comment by tanjunchen** [comment]
Inconsistent with the description in the upgrade text

**Comment by tanjunchen** [comment]
cc @Laci21

**Comment by Laci21** [comment]
@tanjunchen, based on what I can see from your logs, the control plane upgrade went through perfectly.

Please note, that for Istio 1.5 `istiod` was introduced and it is turned on by default, which consists the functionalities of `Citadel`, `Galley`, `Pilot` and the `Sidecar Injector`. That is why at the end you have an `istiod` pod instead of the four aformentioned components.
More info: https://banzaicloud.com/blog/istio-operator-1.5/

For Istio 1.5, the so-called Mixerless telemetry or Telemetry V2 feature became enabled by default as well. This means that no Mixer components will be running on the Istio control plane, this is why the `Policy` and `Telemetry` pods are be deleted from the cluster.
You can read more on this feature here: https://banzaicloud.com/blog/istio-mixerless-telemetry/

Lastly, the egress gateway is disabled by default for 1.5, as it is not always needed for every user. This is why the egress gateway pod is terminated. You can of course enable many egress gateways for your use cases with our operator, you can read more on that here: https://banzaicloud.com/blog/istio-multiple-gateways/

However, you are right that the upgrade docs is inconsistent with the behaviour. The docs should be updated so that it reflects what actually happens and shows correctly the remaining control plane pods after the upgrade flow.

**Comment by Laci21** [comment]
Doc is fixed in #436
